src/components/input/range/index.tsx

```diff
@@ -32,8 +33,10 @@ export class SmoothlyInputRange {
 					type="range"
 					min={this.min}
 					max={this.max}
-					step={"any"}
-					onInput={e => e.target instanceof HTMLInputElement && (this.value = Number.parseFloat(e.target.value))}
+					step={this.step}
+					onInput={e =>
+						e.target instanceof HTMLInputElement && (this.value = Math.trunc(e.target.valueAsNumber * 100) / 100)
```

Comment:
I know i told you trunc earlier but i think rounding is better for this case. I also think we only want to do this if `this.step == "any"`.

Reply:
When a number is passed as "step" there are no decimals so it only has affect on "any" so i thought id save a few lines of code not applying it conditionally. But i can add a condition based on "step" if you want to?

Reply:
step can be a decimal number 
```html
<input type="range" step="0.1" min="0" max="1">
```